### DISCUSSION TITLE
What is the best way to build one target platform at a time?

### DISCUSSION THREAD

**Discussion by TommiGustafsson-HMP** [dotnet/maui#16543] [Q&A] [answered]
A very simple question: **Using the single project model, what is the best way to build one target platform at a time?** Our use case is: First build Android while other platforms are excluded from building. When Android works, build iOS and test. When iOS works, build Windows and test. When Windows works, build Mac Catalyst and test.

While this is possible by editing `TargetFrameworks` the project file, it seems to be quite cumbersome, makes unnecessary changes to the source control, and it also triggers NuGet package restore, and you need to restart Visual Studio, too.

**Comment by mattleibow** [comment, marked as answer]
There is a way and that is to build the project you want using the specific target framework:

```
dotnet build path/to/app.csproj -f net7.0-android
```

Or the other way:

```
dotnet build path/to/app.csproj -p:TargetFramework=net7.0-android
```

**Reply by TommiGustafsson-HMP** [reply]
Thanks Matthew, very much appreciated!

**Comment by munkii** [comment]
How would I do this in Visual Studio 2022 when I am trying to interactively build and deploy to a Device? I feel like this should "Just Work" but it doesn't and I getting iOS compilation errors when I am tyrying to deply to an Android device

**Reply by TommiGustafsson-HMP** [reply]
I believe it is `dotnet build .\myapp.csproj -c:Debug -f:net9.0-android -t:run`

`-t:run` doesn't work on Windows, though.

If you want to rebuild add also `--no-incremental`.

If Android fails to deploy, you can just locate your signed APK file and then use Android ADB Command Prompt and write:

`adb install myapp-Signed.apk`

**Reply by munkii** [reply]
Sorry for the delay in replying. Are you saying that when using the Single Project model with MAUI I cannot simply press "Start Debugging" in the IDE and it just build whatever framework I have currently selected?

**Reply by TommiGustafsson-HMP** [reply]
> Sorry for the delay in replying. Are you saying that when using the Single Project model with MAUI I cannot simply press "Start Debugging" in the IDE and it just build whatever framework I have currently selected?

I think it works, if rebuild is not required. But nowaydays I usually rebuild the project with `dotnet build .\myproject.csproj -f:net9.0-android -c:Debug --no-incremental`. It works for projects that do not depend on Visual Studio only properties in MSBuild.
